src/initilization-context.ts

```diff
@@ -14,23 +14,41 @@ export class InitializationContext {
 		this.valueResolver = valueResolver;
 	}
 
+	/**
+	 * Prevents any waiting callbacks from being executed.
+	 * @returns A callback which removes your delay.
+	 */
+	delayQueue() {
```

Comment:
**Slack conversation**
> taylor  11:55 AM
> I'm just getting around to looking at your feedback on the modeljs PR, re: `delayQueue`. I agree the pattern where you return a "reversal function" is a bit clunky. I'm not opposed to your "resolvable" object idea `let delay = context.delayQueue(); ... ; delay.resolve()`, but just wanted to float another idea:
> ```js
> const delayId = context.delayQueue();
> ...
> context.clearDelay(delayId);
> ```
> 
> bryan  11:55 AM
> Sort of like `setTimeout` and `setInterval`
> 
> taylor  11:55 AM
> Yeah
> 
> bryan  11:57 AM
> Interesting idea. Its kind of reminiscent of asynchronicity, but in this case its always used synchronously, right?
> 
> taylor  11:57 AM
> Yeah, it has to be synchronous
> 
> bryan  11:58 AM
> Is the usage pattern such that it could be something that took a callback, so it was entirely wrapped up in that call? Something like `context.something(() => { do stuff... });`?
> 11:59
> And something would internally set the "delay", execute the callback, then "release" it?
> 
> taylor  12:00 PM
> Hmm interesting idea, let me see
> 12:00
> I think that should work
> 12:01
> It could be an alternate signature for wait
> 
> bryan  12:01 PM
> Yeah, I was wondering the same thing
> 12:02
> The name seems kind of weird though, since its not async
> 
> taylor  12:03 PM
> `context.execute()`?
> 
> bryan  12:04 PM
> Yeah, that's what I was thinking... :thinking_face:

Reply:
I've pushed the refactor of `delayQueue`.

Reply:
LGTM